### DISCUSSION TITLE
plyr hls loads all the ts fragments

### DISCUSSION THREAD

**Discussion by quotesw** [sampotts/plyr#2724] [General]
Hello, I am trying to switch from viblast to plyr.
I am serving HLS livestreams and the recorded event using wowza. 

Using plyr, for the live event is all ok.
Where serving the recorded event (recorder on the wowza) all the ts fragments are downloaded, sometimes few gigabytes, even if is not in play state.

Can I prevent downloading too much data? The preload attribute does nothing. 

With videojs+viblast only few ts files are downloaded.

Any ideas to try?

**Comment by res-io** [comment]
Same here. Did you find a solution yet?

**Reply by quotesw** [reply]
no solution. 

I am still using the **videojs** player.

**Reply by res-io** [reply]
Actually it is not a plyr problem. It is a hls settings.
I jut found a solution by setting the maxBufferSize in the hls constructor:

const hls = new Hls({maxBufferSize: 2 * 1000 * 1000});   //2 for two megabytes default is 60megabytes
hls.loadSource(playerSrc);

This limits the auto download.

Hope this helps.
